packages/regenesis-surgery/scripts/surgery.ts

```diff
@@ -0,0 +1,402 @@
+  // L2StandardBridge
+  '0x4200000000000000000000000000000000000010',
+  // OVM_SequencerFeeVault
+  '0x4200000000000000000000000000000000000011',
+  // L2StandardTokenFactory
+  '0x4200000000000000000000000000000000000012',
+]
+const PREDEPLOY_NO_WIPE_ADDRESSES = [
+  // OVM_L2ToL1MessagePasser
+  '0x4200000000000000000000000000000000000000',
+  // OVM_DeployerWhitelist
+  '0x4200000000000000000000000000000000000002',
+]
+const OLD_ETH_ADDRESS = '0x4200000000000000000000000000000000000006'
+
+const hexStringEqual = (a: string, b: string): boolean => {
+  if (!ethers.utils.isHexString(a)) {
+    throw new Error(`not a hex string: ${a}`)
+  }
+  if (!ethers.utils.isHexString(b)) {
+    throw new Error(`not a hex string: ${b}`)
+  }
+
+  return a.toLowerCase() === b.toLowerCase()
+}
+
+const isEOAContract = (account: Account): boolean => {
+  return EOA_CODE_HASHES.some((codeHash) => {
+    return hexStringEqual(account.codeHash, codeHash)
+  })
+}
+
+const isUniswapV3Factory = (account: Account): boolean => {
+  return hexStringEqual(account.address, UNISWAP_V3_FACTORY_ADDRESS)
+}
+
+const isUniswapV3NFPM = (account: Account): boolean => {
+  return hexStringEqual(account.address, UNISWAP_V3_NFPM_ADDRESS)
+}
+
+const isUniswapPool = (account: Account, data: SurgeryDataSources): boolean => {
+  return data.pools.some((pool) => {
+    return hexStringEqual(pool.oldAddress, account.address)
+  })
+}
+
+const isUniswapLibrary = (account: Account): boolean => {
+  return UNISWAP_V3_LIB_ADDRESSES.some((addr) => {
+    return hexStringEqual(account.address, addr)
+  })
+}
+
+const isUniswapContract = (account: Account): boolean => {
+  return UNISWAP_V3_CONTRACT_ADDRESSES.some((addr) => {
+    return hexStringEqual(account.address, addr)
+  })
+}
+
+const isPrecompile = (account: Account): boolean => {
+  return account.address
+    .toLowerCase()
+    .startsWith('0x00000000000000000000000000000000000000')
+}
+
+const isPredeployWipe = (account: Account): boolean => {
+  return PREDEPLOY_WIPE_ADDRESSES.some((addr) => {
+    return hexStringEqual(account.address, addr)
+  })
+}
+
+const isPredeployNoWipe = (account: Account): boolean => {
+  return PREDEPLOY_NO_WIPE_ADDRESSES.some((addr) => {
+    return hexStringEqual(account.address, addr)
+  })
+}
+
+const isPredeployDead = (account: Account): boolean => {
+  const PREDEPLOY_DEAD_ADDRESSES = [
+    // OVM_L1MessageSender
+    '0x4200000000000000000000000000000000000001',
+  ]
+
+  return PREDEPLOY_DEAD_ADDRESSES.some((addr) => {
+    return hexStringEqual(account.address, addr)
+  })
+}
+
+const isPredeployETH = (account: Account): boolean => {
+  return hexStringEqual(
+    account.address,
+    '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000'
+  )
+}
+
+const isPredeployWETH = (account: Account): boolean => {
+  return hexStringEqual(account.address, OLD_ETH_ADDRESS)
+}
+
+const isEtherscanVerified = (account: Account): boolean => {
+  return false // TODO
```

Comment:
This is going to need a second argument or refactored into an object oriented style where the etherscan dump can be accessed via `this`